evi-python-function-calling/README.md

```diff
@@ -0,0 +1,161 @@
+<div align="center">
+  <img src="https://storage.googleapis.com/hume-public-logos/hume/hume-banner.png">
+  <h1>EVI Python Function Calling Example</h1>
+</div>
+
+## Overview
+
+This project showcases how to call functions in a sample implementation of Hume's [Empathic Voice Interface](https://hume.docs.buildwithfern.com/docs/empathic-voice-interface-evi/overview) using Hume's Python SDK. Here, we have a simple EVI that calls a function to get the current weather for a given location.
+
+## Prerequisites
+
+The Hume Python SDK supports Python versions 3.9, 3.10, and 3.11 on macOS and Linux systems.
+
+### Setting up a virtual environment (optional)
+
+You can create a virtual environment using either `conda` from Miniconda or Anaconda, or the built-in `venv` module in Python. Instructions for both methods are provided below.
+
+#### Using `conda`
+1. Install `conda` using [Miniconda](https://docs.anaconda.com/miniconda/), the free minimal installer for it.
+2. Create a new virtual environment:
+    ```bash
+    conda create --name evi-env python=3.11
+    ```
+3. Activate the virtual environment:
+    ```bash
+    conda activate evi-env
+    ```
+
+#### Using `venv`
+
+To create a virtual environment with `venv`, run the following commands in your terminal:
+
+1. Create a new virtual environment:
+    ```bash
+    python -m venv evi-env
+    ```
+2. Activate the virtual environment:
+    ```bash
+    source evi-env/bin/activate
+    ```
+
+### Package and system dependencies
+
+Follow these steps to install the required packages and system dependencies for using environment variables, EVI, and handling audio input/output.
+
+#### Environment Variables
+
+Install the `python-dotenv` package to load variables from a `.env` file:
+
+```bash
+pip install python-dotenv
+```
+
+#### Asynchronous HTTP requests
+
+Install the `aiohttp` package to make asynchronous HTTP requests in the weather fetching function:
+
+```bash
+pip install aiohttp
```

Comment:
Any reason for `aiohttp` instead of `httpx` like we use as the http client inside hume-python-sdk?

Reply:
While developing the example, I did some research and `aiohttp` stood out as a really efficient asynchronous HTTP client. That said, you're absolutely right that `httpx` is effectively included for free with `pip install hume`. I'll refactor the script to use `httpx` instead.